pkg/apis/core/validation/validation.go

```diff
@@ -1491,6 +1501,15 @@ func validateCSIPersistentVolumeSource(csi *core.CSIPersistentVolumeSource, fldP
 	return allErrs
 }
 
+func validateCSIVolumeSource(csi *core.CSIVolumeSource, fldPath *field.Path) field.ErrorList {
```

Comment:
validation seems underspecified
* should driver name be validated (max length, allowable format)
* volume handle max length, allowable format
* the API doc indicates VolumeHandle is required. is that where the discussion about per-pod volumes ended up? either validate it as required here or fix the API doc
* if secret references are non-nil, require name to be present and a well-formed secret name

Reply:
@liggitt 
> the API doc indicates VolumeHandle is required. is that where the discussion about per-pod volumes ended up? either validate it as required here or fix the API doc
I will fix the API doc.  Where we ended up is that it will not be required for inline per-vol pod.

Reply:
sounds good. be sure to add `+optional` and `omitempty` as appropriate